app/src/main/java/com/example/bigowlapp/activity/HomePageActivity.java

```diff
@@ -136,6 +145,25 @@ private void subscribeToData() {
          */
     }
 
+    /**
+     * Initializes the AlarmManager to set alarms for the user's schedules that he has to attend to.
+     */
+    private void initAlarmManager() {
+        if (alarmBroadcastReceiverManager == null) {
+            alarmBroadcastReceiverManager = new AlarmBroadcastReceiverManager(this);
+        }
+        alarmBroadcastReceiverManager.setAlarms(homePageViewModel.getCurrentUserUid());
```

Comment:
Good use of a manager so that the activity isn't bloated, and we can easily move it to other places if needed